### PR TITLE
Update .editorconfig to match file from coreclr

### PR DESCRIPTION
**Pull request by KindDragon**
Changed to match
* Roslyn https://github.com/dotnet/roslyn/blob/master/.editorconfig
* CoreClr https://github.com/dotnet/coreclr/blob/master/.editorconfig

### CODE REVIEW

**Comment by mellinoe** [comment]
LGTM

File: .editorconfig [hunk 1]
```diff
@@ -6,19 +6,20 @@ root = true
 
 # Default settings:
 # A newline ending every file
-# Use 2 spaces as indentation
-# Trim trailing whitespace
+# Use 4 spaces as indentation
 [*]
 insert_final_newline = true
```
Comment (sharwell) [inline .editorconfig]: {F702, layout=inline, width=20} This line should be removed IMO. It behaves "poorly" when used with certain designers and the project system and doesn't offer much.

Comment (KindDragon) [inline .editorconfig]: Linux tools prefer final new line http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline

Comment (ellismg) [inline .editorconfig]: I think the place we are moving towards is to have the file have a final newline when it's human generated and just accept whatever tool generated files generate.

I don't have a strong preference either way, but Git and GitHub seem to prefer having the additional newline, so I'm fine with adding it when it doesn't require gymnastics.

**Comment by ellismg** [comment]
LGTM.  Thanks.

**Pull request merged**